src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -0,0 +1,278 @@
+   * @param conductivity Thermal conductivity of the material (power / (length * temp))
+   */
+  LinearIsotropicConductor(double density = 1.0, double specific_heat_capacity = 1.0, double conductivity = 1.0)
+      : density_(density), specific_heat_capacity_(specific_heat_capacity), conductivity_(conductivity)
+  {
+    SLIC_ERROR_ROOT_IF(conductivity_ < 0.0,
+                       "Conductivity must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(specific_heat_capacity_ < 0.0,
+                       "Specific heat capacity must be positive in the linear isotropic conductor material model.");
+  }
+
+  /**
+   * @brief Function defining the thermal flux (constitutive response)
+   *
+   * @tparam T1 type of the temperature (e.g. tensor or dual type)
+   * @tparam T2 type of the temperature gradient (e.g. tensor or dual type)
+   * @param temperature_gradient Gradient of the temperature (du_dx)
+   * @return The thermal flux of the material model
+   */
+  template <typename T1, typename T2>
+  SERAC_HOST_DEVICE T2 operator()(T1& /* temperature */, T2& temperature_gradient) const
+  {
+    return -1.0 * conductivity_ * temperature_gradient;
+  }
+
+  /**
+   * @brief The density (mass per volume) of the material model
+   *
+   * @tparam dim The dimension of the problem
+   * @return The density
+   */
+  template <int dim>
+  SERAC_HOST_DEVICE double density(tensor<double, dim>& /* x */) const
+  {
+    return density_;
+  }
+
+  /**
+   * @brief The specific heat capacity (heat capacity per unit mass) of the material model
+   *
+   * @tparam dim The dimension of the problem
+   * @tparam T Type of the temperature variable
+   */
+  template <typename T, int dim>
+  SERAC_HOST_DEVICE double specificHeatCapacity(tensor<double, dim>& /* x */, T& /* temperature */) const
+  {
+    return specific_heat_capacity_;
+  }
+
+private:
+  /// Density
+  double density_ = 1.0;
+
+  /// Specific heat capacity
+  double specific_heat_capacity_ = 1.0;
+
+  /// Constant isotropic thermal conductivity
+  double conductivity_ = 1.0;
+};
+
+/**
+ * @brief Linear anisotropic thermal material model
+ *
+ * @tparam dim Spatial dimension
+ */
+template <int dim>
+class LinearConductor {
+public:
+  /**
+   * @brief Construct a new Linear Isotropic Conductor object
+   *
+   * @param density Density of the material (mass/volume)
+   * @param specific_heat_capacity Specific heat capacity of the material (energy / (mass * temp))
+   * @param conductivity Thermal conductivity of the material (power / (length * temp))
+   */
+  LinearConductor(double density = 1.0, double specific_heat_capacity = 1.0,
+                  tensor<double, dim, dim> conductivity = {{{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}}})
```

Comment:
Does `tensor<double, dim, dim> conductivity = {{{1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0}}}` compile when dim == 2?

```suggestion
                  tensor<double, dim, dim> conductivity = Identity<dim>())
```

Reply:
Nope! Thanks for the catch!